File: src/c#/Proxy/VekilBasbakan.cs
using System;
namespace DesignPatterns.org.javaturk.dp.basic.pattern.gof.structural.proxy.basbakan
{
    public class VekilBasbakan : IBasbakan
    {
        private IBasbakan gercekBasbakan;

        public VekilBasbakan(IBasbakan gercekBasbakan)
        {
            this.gercekBasbakan = gercekBasbakan;
        }

        public void DertDinle(String dert)
        {
            Console.WriteLine("\nVekil: Derdinizi dinliyorum.");

            //      if(......)  
            bool ayiklandi = Ayikla(dert);
            if (ayiklandi)
                Ilet(dert);
        }

        public void IsBul(String yakinim)
        {
            Console.WriteLine("\nVekil: Isteginizi dinliyorum.");
        }

        private bool Ayikla(String dert)
        {
            bool b = true;
            // Bütün kontroller burada!
            return b;
        }

        private void Ilet(String dert)
        {
            gercekBasbakan.DertDinle(dert);
        }
    }
}

File: src/java/org/javaturk/dp/ch05/proxy/basbakan/VekilBasbakan.java
package org.javaturk.dp.ch05.proxy.basbakan;

public class VekilBasbakan implements Basbakan {
	
	private Basbakan gercekBasbakan;
	
	public VekilBasbakan(Basbakan gercekBasbakan){
		this.gercekBasbakan = gercekBasbakan;
	}

	@Override
	public void dertDinle(String dert) {
		System.out.println("Vekil: Derdinizi dinliyorum.");
		
//		if(......)  // Bütün kontroller burada!
		boolean ayiklandi = ayikla(dert);
		if(ayiklandi)
			ilet(dert);
	}

	@Override
	public void isBul(String yakinim) {
		System.out.println("Vekil: Isteginizi dinliyorum.");
	}
	
	private boolean ayikla(String dert){
		boolean b = true;
		//...
		return b;
	}
	
	private void ilet(String dert){
		gercekBasbakan.dertDinle(dert);
	}
}
